vulntotal/tests/test_snyk.py

```diff
@@ -52,6 +52,52 @@ def test_generate_package_advisory_url(self):
         ]
         util_tests.check_results_against_expected(results, expected)
 
+    def test_generate_purl(self):
+        package_advisory_urls = [
+            "https://security.snyk.io/package/pip/jinja2",
+            "https://security.snyk.io/package/maven/org.apache.tomcat%3Atomcat",
+            "https://security.snyk.io/package/npm/semver-regex",
+            "https://security.snyk.io/package/npm/@urql%2Fnext",
+            "https://security.snyk.io/package/npm/@lobehub%2Fchat",
+            "https://security.snyk.io/package/npm/meshcentral",
+            "https://security.snyk.io/package/composer/bolt%2Fcore",
+            "https://security.snyk.io/package/linux/debain:11/trafficserver",
+            "https://security.snyk.io/package/linux/almalinux:8/rpm-plugin-fapolicyd",
+            "https://security.snyk.io/package/nuget/moment.js",
+            "https://security.snyk.io/package/cocoapods/ffmpeg",
+            "https://security.snyk.io/package/hex/coherence",
+            "https://security.snyk.io/package/rubygems/log4j-jars",
+            "https://security.snyk.io/package/golang/github.com%2Fgrafana%2Fgrafana%2Fpkg%2Fservices%2Fsqlstore%2Fmigrator",
+            "https://security.snyk.io/package/golang/github.com%2Fanswerdev%2Fanswer%2Finternal%2Frepo%2Factivity",
+            "https://security.snyk.io/package/golang/github.com%2Fargoproj%2Fargo-cd%2Fv2%2Futil%2Fhelm",
```

Comment:
For better test coverage, add few non-GitHub Go packages in the test too.

Reply:
Done, added a few non Github Go packages